File: DataStructure/10828_스택.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

void solution(void) {
	int T;
	stack<int> s;
	cin >> T;
	while (T--) {
		string cmm;
		int value;
		cin >> cmm;

		if (cmm == "push") {
			cin >> value;
			s.push(value);
		}
		else if (cmm == "pop") {
			if (s.empty())
				cout << -1 << '\n';
			else {
				cout << s.top() << '\n';
				s.pop();
			}
		}

		else if (cmm == "size")
			cout << s.size() << '\n';

		else if (cmm == "empty") {
			if (s.empty())
				cout << 1 << '\n';
			else
				cout << 0 << '\n';
		}

		else if (cmm == "top") {
			if (s.empty())
				cout << -1 << '\n';
			else
				cout << s.top() << '\n';
		}
	}
	return;
}

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	solution();
	return 0;
}

File: DataStructure/10828_스택.py
#입출력
from sys import stdin
#입력 
input = stdin.readline

#테스트 입력
T=int(input())
stack=[]

for _ in range(T):
    #명령어 입력
    comm=list(input().split())
    #push
    if comm[0] == "push":
        stack.append(int(comm[1]))
    
    #top
    elif comm[0] == "top":
        print(stack[-1] if len(stack)>0 else -1)

    #size
    elif comm[0] == "size":
        print(len(stack))
    
    #empty
    elif comm[0] == "empty":
        print(1 if len(stack)==0 else 0)
    
    #pop
    elif comm[0] == "pop":
        print(stack.pop() if len(stack)>0 else -1)